### ISSUE TITLE
PowerWalker VI 1200 SHL regressions in NUT 2.8.0

### ISSUE THREAD

**Issue by blazzakrajsek** [networkupstools/nut#1967]
I have noticed that upgrade of Debian 11 (bullseye) to Debian 12 (bookworm) upgraded the `nut` package from 2.7.4 to 2.8.0.
The support for PowerWalker VI 1200 SHL is now worse, see my previous report #1270. I have noticed that the driver has changed from `MGE HID 1.40` to `Phoenixtec/Liebert HID 0.41`.

Current status under NUT 2.8.0 for PowerWalker VI 1200 SHL is as follows...
`ups.conf`:
```
[ups0]
    driver = "usbhid-ups"
    port = "auto"
    vendorid = "06DA"
    productid = "FFFF"
```

`upsc ups0`:
```
Init SSL without certificate database
battery.charge: 100
battery.runtime: 7200
battery.type: PbAc
battery.voltage: 27.00
device.mfr: PPC
device.model: Offline UPS
device.serial: 000000000   
device.type: ups
driver.name: usbhid-ups
driver.parameter.pollfreq: 30
driver.parameter.pollinterval: 2
driver.parameter.port: auto
driver.parameter.productid: FFFF
driver.parameter.synchronous: auto
driver.parameter.vendorid: 06DA
driver.version: 2.8.0
driver.version.data: Phoenixtec/Liebert HID 0.41
driver.version.internal: 0.47
driver.version.usb: libusb-1.0.26 (API: 0x1000109)
input.voltage: 237.0
output.frequency: 49.80
output.voltage: 237.0
ups.load: 0
ups.mfr: PPC
ups.model: Offline UPS
ups.productid: ffff
ups.serial: 000000000   
ups.status: OL CHRG
ups.vendorid: 06da
```

`upsrw -l ups0`:
```
(empty)
```

`upscmd -l ups0`:
```
Instant commands supported on UPS [ups0]:
(empty)
```

Commands and variables are now not available any more. For example, with the `shutdown.return` command missing, UPS shutdown is not possible.

I guess (as I did not test) that the fixes implemented by #1357 made support for Phoenixtec 0x06DA:0xFFFF devices in general poorer.

**Comment by jimklimov** [comment]
Thanks for the report! it would really be ibteresting to know if that PR is the culprit - e.g. if its disablement for non-AEG branded devices should be expanded to more names, or made a config toggle...

**Comment by blazzakrajsek** [comment]
Glad to help. Indeed the commit `d30ffffb8f` changed the driver for this device.

---
### `d30ffffb8f`:

`upsc ups0`: 
``` 
battery.charge: 100
battery.runtime: 7200
battery.type: PbAc
battery.voltage: 27.00
device.mfr: PPC
device.model: Offline UPS
device.serial: 000000000
device.type: ups
driver.name: usbhid-ups
driver.parameter.pollfreq: 30
driver.parameter.pollinterval: 2
driver.parameter.port: auto
driver.parameter.productid: FFFF
driver.parameter.synchronous: auto
driver.parameter.vendorid: 06DA
driver.version: 2.8.0-rc1-34-gd30ffffb8
driver.version.data: Phoenixtec/Liebert HID 0.4
driver.version.internal: 0.46
driver.version.usb: libusb-1.0.26 (API: 0x1000109)
input.voltage: 238.0
output.frequency: 49.80
output.voltage: 238.0
ups.load: 0
ups.mfr: PPC
ups.model: Offline UPS
ups.productid: ffff
ups.serial: 000000000
ups.status: CHRG
ups.vendorid: 06da
```

`upsrw ups0`:
```
(empty)
```

`upscmd -l ups0`:
```
Instant commands supported on UPS [ups0]:
(empty)
```

---
### `51e6b886a`:

`upsc ups0`:
```
battery.charge: 100
battery.runtime: 7200
battery.type: PbAc
device.mfr: PPC
device.model: Offline UPS HID UPS
device.serial: 000000000   
device.type: ups
driver.name: usbhid-ups
driver.parameter.pollfreq: 30
driver.parameter.pollinterval: 2
driver.parameter.port: auto
driver.parameter.productid: FFFF
driver.parameter.synchronous: auto
driver.parameter.vendorid: 06DA
driver.version: 2.8.0-rc1-33-g51e6b886a
driver.version.data: MGE HID 1.45
driver.version.internal: 0.46
driver.version.usb: libusb-1.0.26 (API: 0x1000109)
input.voltage: 237.0
outlet.1.status: on
output.frequency: 49.6
output.frequency.nominal: 50
output.voltage: 237.0
output.voltage.nominal: 230
ups.beeper.status: enabled
ups.delay.shutdown: 20
ups.delay.start: 30
ups.load: 0
ups.mfr: PPC
ups.model: Offline UPS HID UPS
ups.power.nominal: 5
ups.productid: ffff
ups.realpower: 0
ups.serial: 000000000   
ups.status: 
ups.test.result: No test initiated
ups.timer.shutdown: -1
ups.timer.start: -1
ups.type: offline / line interactive
ups.vendorid: 06da
```

`upsrw ups0`:
```
[output.voltage.nominal]
Nominal output voltage (V)
Type: ENUM STRING
Maximum length: 3
Option: "200"
Option: "208"
Option: "220"
Option: "230" SELECTED
Option: "240"

[ups.delay.shutdown]
Interval to wait after shutdown with delay command (seconds)
Type: STRING
Maximum length: 10
Value: 20

[ups.delay.start]
Interval to wait before (re)starting the load (seconds)
Type: STRING
Maximum length: 10
Value: 30
```

`upscmd -l ups0`:
```
Instant commands supported on UPS [ups0]:

beeper.disable - Disable the UPS beeper
beeper.enable - Enable the UPS beeper
beeper.mute - Temporarily mute the UPS beeper
beeper.off - Obsolete (use beeper.disable or beeper.mute)
beeper.on - Obsolete (use beeper.enable)
load.off - Turn off the load immediately
load.off.delay - Turn off the load with a delay (seconds)
load.on - Turn on the load immediately
load.on.delay - Turn on the load with a delay (seconds)
shutdown.return - Turn off the load and return when power is back
shutdown.stayoff - Turn off the load and remain off
shutdown.stop - Stop a shutdown in progress
test.battery.start.deep - Start a deep battery test
test.battery.start.quick - Start a quick battery test
test.battery.stop - Stop the battery test
```

---
However, please note that with the changes that I proposed in #1968 only the features actually supported by the hardware are available. For example, beeper can only be toggled and battery test only has a test option (no quick, deep and stop options) as far as I have tested the commands.

I have also added this:
```
  ...
  { "battery.charge.capacity.design", 0, 0, "UPS.PowerSummary.DesignCapacity", NULL, "%.0f", 0, NULL },
  { "battery.charge.capacity.remaining", 0, 0, "UPS.PowerSummary.FullChargeCapacity", NULL, "%.0f", 0, NULL },
```
These names are not documented in nut-names.txt, and can either be added to the documentation, removed or renamed if you have any better suggestions. :)

---
For reference, with the changes proposed in #1968 it is as follows...
`upsc ups0`:
```
battery.charge: 100
battery.charge.capacity.design: 100
battery.charge.capacity.remaining: 100
battery.runtime: 7200
battery.type: PbAc
battery.voltage: 27.00
device.mfr: PPC
device.model: Offline UPS
device.serial: 000000000   
device.type: ups
driver.debug: 6
driver.flag.allow_killpower: 0
driver.name: usbhid-ups
driver.parameter.pollfreq: 30
driver.parameter.pollinterval: 2
driver.parameter.port: auto
driver.parameter.productid: FFFF
driver.parameter.synchronous: auto
driver.parameter.vendorid: 06DA
driver.state: updateinfo
driver.version: 2.8.0-2193-g02eea9cbc
driver.version.data: Phoenixtec/Liebert HID 0.41
driver.version.internal: 0.49
driver.version.usb: libusb-1.0.26 (API: 0x1000109)
input.voltage: 239.0
output.frequency: 49.80
output.frequency.nominal: 50
output.voltage: 239.0
output.voltage.nominal: 230
ups.beeper.status: enabled
ups.delay.shutdown: 20
ups.delay.start: 30
ups.load: 0
ups.mfr: PPC
ups.model: Offline UPS
ups.power.nominal: 5
ups.productid: ffff
ups.serial: 000000000   
ups.status: OL CHRG
ups.test.result: No test initiated
ups.vendorid: 06da
```

`upsrw ups0`:
```
[driver.debug]
Description unavailable
Type: NUMBER
Value: 6

[driver.flag.allow_killpower]
Description unavailable
Type: NUMBER
Value: 0

[ups.delay.shutdown]
Interval to wait after shutdown with delay command (seconds)
Type: STRING
Maximum length: 10
Value: 20

[ups.delay.start]
Interval to wait before (re)starting the load (seconds)
Type: STRING
Maximum length: 10
Value: 30
```

`upscmd -l ups0`:
```
Instant commands supported on UPS [ups0]:

beeper.toggle - Toggle the UPS beeper
driver.killpower - Description unavailable
driver.reload - Description unavailable
driver.reload-or-error - Description unavailable
driver.reload-or-exit - Description unavailable
load.off - Turn off the load immediately
load.off.delay - Turn off the load with a delay (seconds)
load.on - Turn on the load immediately
load.on.delay - Turn on the load with a delay (seconds)
shutdown.return - Turn off the load and return when power is back
shutdown.stayoff - Turn off the load and remain off
shutdown.stop - Stop a shutdown in progress
test.battery.start - Start a battery test
```

As you suggested, I believe the best course of action would be to add a subdriver or some other option to be able to better control the options available for the exact UPS, since the VID/PID 0x06DA:0xFFFF is obviously used by many different devices.
I am open to suggestions and let me know if I can help any further. :)

**Comment by jimklimov** [comment]
Noting for posterity: some progress about #1369 (essentially, adding `subdriver=...` support to `usbhid-ups`) could also help here.

**Comment by blazzakrajsek** [comment]
I have renamed the ticket to better reflect the issue. Closing issue as resolved with #1968.
